kjourney/src/app/data.service.ts

```diff
@@ -28,4 +30,12 @@ export class DataService {
   public getResponsibilities(bandID): Observable<Responsibility[]> {
     return this.http.get<Responsibility[]>('api/responsibilities/' + bandID);
   }
+
+  public getBandRoles(bandID): Observable<BandRole[]> {
+    return this.http.get<BandRole[]>('api/roles/' + bandID);
```

Comment:
you are getting roles by bandId so you need to use query parameters 

`api/roles/?bandId=`